jetty-websocket/jetty-websocket-common/src/main/java/org/eclipse/jetty/websocket/common/JettyWebSocketFrameHandler.java

```diff
@@ -435,6 +435,8 @@ public void resume()
 
         if (delayedFrame != null)
             delayedFrame.run();
+        else
+            demand();
```

Comment:
why call demand and grab the lock again?  Better to decide what to do in the lock above

Reply:
note also, in `demand()` and in this method when updated, do not call `demand(1)` from within lock block.